src/WinGetStudio/Assets/Monaco/editor.html

```diff
@@ -51,9 +58,30 @@
                         case 'setReadOnly':
                             editor.updateOptions({ readOnly: msg.value || false });
                             break;
+                        case 'setCodeLenses':
+                            SetCodeLense(msg.value);
+                            break;
+                        case 'setCodeLensesEnabled':
+                            editor.updateOptions({ codeLens: msg.value || false });
+                            break;
                     }
                 });
             }
+
+            // CodeLens helper function for updating lenses
+            let currentProviderDisposable = null;
+            function SetCodeLense(serializedLenses) {
+                try {
+                    let lenses = JSON.parse(serializedLenses);
+                    currentProviderDisposable?.dispose();
```

Comment:
I'm assuming this is an established Monaco pattern, but it looks like strange usage and want to confirm this is correct. If this is a disposable resource, is "using" not appropriate here instead of calling dispose the next time time "SetCodeLens" is called? Appears like we are using this like a static variable and keeping around the latest copy and disposing of the old one every time we call this. Is that correct for this usage?

Reply:
Correct! Here `SetCodeLenses` is called from C# to set/replace the codelenses for the Monaco editor. To do that, we call dispose on the previous provider and register a new one which will result in replacing the old codelenses with the new ones.